### PR TITLE
feat: add option to return disabled strategies

### PR DESCRIPTION
**Pull request by andreas-unleash**
Adds the option to include disabled strategies (behind the playgroundImprovements flag

Closes # [1-1505](https://linear.app/unleash/issue/1-1505/return-disabled-strategies-in-the-playground-features-request)

### CODE REVIEW

File: src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts [hunk 1]
```diff
@@ -123,7 +125,10 @@ export class FeatureToggleRowConverter {
         feature.variants = row.variants || [];
         feature.project = row.project;
 
-        if (this.isUnseenStrategyRow(feature, row) && !row.strategy_disabled) {
+        if (
+            this.isUnseenStrategyRow(feature, row) &&
+            (includeDisabledStrategies ? true : !row.strategy_disabled)
+        ) {
```
Comment (FredrikOseberg) [inline src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts:131]: I don't think this is the correct place to put this. createBaseFeature should only care about what is common between the two API endpoints. Instead you should take this check and move into the respective methods: `buildFeatureToggleListFromRows` and `buildPlaygroundFeaturesFromRows` and have each method have different checks for which strategies they allow

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
We should improve the architecture slightly before merging

File: src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts (hunk 1 above)
Comment (andreas-unleash) [reply src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts:131]: Ok makes sense

File: src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts [hunk 2]
```diff
@@ -149,6 +151,14 @@ export class FeatureToggleRowConverter {
...
             feature = this.createBaseFeature(r, feature, featureQuery);
 
+            if (
+                this.isUnseenStrategyRow(feature, r) &&
+                includeDisabledStrategies &&
+                r.strategy_disabled
+            ) {
+                feature.strategies?.push(this.rowToStrategy(r));
+            }
+
```
Comment (FredrikOseberg) [inline src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts:161]: Why does the feature toggle list need disabled strategies?

**Review by FredrikOseberg — CHANGES_REQUESTED** [review]
I think this is only relevant to the playground API

File: src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts (hunk 2 above)
Comment (andreas-unleash) [reply src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts]: Following to the frontend - the disabled strategies are not needed

File: src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts (hunk 1 above)
Comment (andreas-unleash) [reply src/lib/features/feature-toggle/converters/feature-toggle-row-converter.ts:131]: Turns out that both of those functions are only used for the Admin API - for which we (everywhere) return the disabled strategies

**Comment by andreas-unleash** [comment]
Not only relevant to playground. To be more specific - this is needed for the admin api

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Pull request merged**